### PR TITLE
Fix moving nested type to another file from an interface

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/63114
Fixes: https://github.com/dotnet/roslyn/issues/60708

### CODE REVIEW

**Review by Youssef1313 — APPROVED** [review]
Does this fix https://github.com/dotnet/roslyn/issues/60708 as well?

**Comment by DoctorKrolic** [comment]
Fixes: https://github.com/dotnet/roslyn/issues/60708

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -764,6 +764,10 @@ internal override SyntaxNode AsInterfaceMember(SyntaxNode m)
                 Accessibility acc;
                 DeclarationModifiers modifiers;
 
+                // return any nested member "as is" without any additional changes
+                if (member is BaseTypeDeclarationSyntax)
+                    return member;
```
Comment (Youssef1313) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:769]: Not sure if it's more safe to `return member` in the default case of switch below. @CyrusNajmabadi What do you think?

Comment (Youssef1313) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:769]: We might have an incomplete member for example.

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:769]: i think this is ok.  IncompleteMember will likely need to be handled differently.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**